### PR TITLE
Fix strlen

### LINKED ISSUE #555: "strlen() expects parameter 1 to be string, object given" at symfony-cmf/cmf-phpcr-dbal-pack recipe install
#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
sonata-project/block-bundle        3.14.0 3.14.0 Symfony SonataBlockBundle
sonata-project/cache               2.0.1  2.0.1  Cache library
sonata-project/core-bundle         3.15.1 3.15.1 Symfony SonataCoreBundle
sonata-project/datagrid-bundle     2.4.0  2.4.0  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions 1.1.5  1.1.5  Doctrine2 behavioral extensions
```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
symfony/asset                v4.2.3  v4.2.3  Symfony Asset Component
symfony/browser-kit          v4.2.3  v4.2.3  Symfony BrowserKit Component
symfony/cache                v4.2.3  v4.2.3  Symfony Cache component with PSR-6, PSR-16, and tags
symfony/config               v4.2.3  v4.2.3  Symfony Config Component
symfony/console              v4.2.3  v4.2.3  Symfony Console Component
symfony/contracts            v1.0.2  v1.0.2  A set of abstractions extracted out of the Symfony components
symfony/css-selector         v4.2.3  v4.2.3  Symfony CssSelector Component
symfony/debug                v4.2.3  v4.2.3  Symfony Debug Component
symfony/debug-bundle         v4.2.3  v4.2.3  Symfony DebugBundle
symfony/debug-pack           v1.0.7  v1.0.7  A debug pack for Symfony projects
symfony/dependency-injection v4.2.3  v4.2.3  Symfony DependencyInjection Component
symfony/doctrine-bridge      v4.2.3  v4.2.3  Symfony Doctrine Bridge
symfony/dom-crawler          v4.2.3  v4.2.3  Symfony DomCrawler Component
symfony/dotenv               v4.2.3  v4.2.3  Registers environment variables from a .env file
symfony/event-dispatcher     v4.2.3  v4.2.3  Symfony EventDispatcher Component
symfony/expression-language  v4.2.3  v4.2.3  Symfony ExpressionLanguage Component
symfony/filesystem           v4.2.3  v4.2.3  Symfony Filesystem Component
symfony/finder               v4.2.3  v4.2.3  Symfony Finder Component
symfony/flex                 v1.1.8  v1.1.8  Composer plugin for Symfony
symfony/form                 v4.2.3  v4.2.3  Symfony Form Component
symfony/framework-bundle     v4.2.3  v4.2.3  Symfony FrameworkBundle
symfony/http-foundation      v4.2.3  v4.2.3  Symfony HttpFoundation Component
symfony/http-kernel          v4.2.3  v4.2.3  Symfony HttpKernel Component
symfony/inflector            v4.2.3  v4.2.3  Symfony Inflector Component
symfony/intl                 v4.2.3  v4.2.3  A PHP replacement layer for the C intl extension that includes addit...
symfony/maker-bundle         v1.11.3 v1.11.3 Symfony Maker helps you create empty commands, controllers, form cla...
symfony/monolog-bridge       v4.2.3  v4.2.3  Symfony Monolog Bridge
symfony/monolog-bundle       v3.3.1  v3.3.1  Symfony MonologBundle
symfony/options-resolver     v4.2.3  v4.2.3  Symfony OptionsResolver Component
symfony/orm-pack             v1.0.6  v1.0.6  A pack for the Doctrine ORM
symfony/panther              v0.2.0  v0.2.0  A browser testing and web scraping library for PHP and Symfony.
symfony/phpunit-bridge       v4.2.3  v4.2.3  Symfony PHPUnit Bridge
symfony/polyfill-intl-icu    v1.10.0 v1.10.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring    v1.10.0 v1.10.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72       v1.10.0 v1.10.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP ver...
symfony/process              v4.2.3  v4.2.3  Symfony Process Component
symfony/profiler-pack        v1.0.4  v1.0.4  A pack for the Symfony web profiler
symfony/property-access      v4.2.3  v4.2.3  Symfony PropertyAccess Component
symfony/property-info        v4.2.3  v4.2.3  Symfony Property Info Component
symfony/routing              v4.2.3  v4.2.3  Symfony Routing Component
symfony/security-bundle      v4.2.3  v4.2.3  Symfony SecurityBundle
symfony/security-core        v4.2.3  v4.2.3  Symfony Security Component - Core Library
symfony/security-csrf        v4.2.3  v4.2.3  Symfony Security Component - CSRF Library
symfony/security-guard       v4.2.3  v4.2.3  Symfony Security Component - Guard
symfony/security-http        v4.2.3  v4.2.3  Symfony Security Component - HTTP Integration
symfony/serializer           v4.2.3  v4.2.3  Symfony Serializer Component
symfony/serializer-pack      v1.0.2  v1.0.2  A pack for the Symfony serializer
symfony/stopwatch            v4.2.3  v4.2.3  Symfony Stopwatch Component
symfony/swiftmailer-bundle   v3.2.5  v3.2.5  Symfony SwiftmailerBundle
symfony/templating           v4.2.3  v4.2.3  Symfony Templating Component
symfony/test-pack            v1.0.5  v1.0.5  A pack for functional and end-to-end testing within a Symfony app
symfony/translation          v4.2.3  v4.2.3  Symfony Translation Component
symfony/twig-bridge          v4.2.3  v4.2.3  Symfony Twig Bridge
symfony/twig-bundle          v4.2.3  v4.2.3  Symfony TwigBundle
symfony/validator            v4.2.3  v4.2.3  Symfony Validator Component
symfony/var-dumper           v4.2.3  v4.2.3  Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter         v4.2.3  v4.2.3  A blend of var_export() + serialize() to turn any serializable data ...
symfony/web-link             v4.2.3  v4.2.3  Symfony WebLink Component
symfony/web-profiler-bundle  v4.2.3  v4.2.3  Symfony WebProfilerBundle
symfony/web-server-bundle    v4.2.3  v4.2.3  Symfony WebServerBundle
symfony/yaml                 v4.2.3  v4.2.3  Symfony Yaml Component
```

#### PHP version

```
$ php -v
PHP 7.2.14-1+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Jan 13 2019 10:05:45) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.14-1+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
```

## Subject
BlockBundle causes error "strlen() expects parameter 1 to be string, object given in block-bundle/src/DependencyInjection/Compiler/TweakCompilerPass.php:117" at `symfony-cmf/cmf-phpcr-dbal-pack` recipe install. This detected on vanilla actual Symfony version (4.2.3).

## Steps to reproduce
```
symfony new --full phpcr-hello
cd phpcr-hello
composer require symfony-cmf/cmf-phpcr-dbal-pack
```
## Expected results
Installing `symfony-cmf/cmf-phpcr-dbal-pack` recipe successfully.
## Actual results

```> composer require symfony-cmf/cmf-phpcr-dbal-pack
Using version ^1.0 for symfony-cmf/cmf-phpcr-dbal-pack
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Restricting packages listed in "symfony/symfony" to "4.2.*"

Prefetching 9 packages 🎶
  - Downloading (100%)

Package operations: 26 installs, 0 updates, 0 removals
  - Installing symfony-cmf/routing (2.1.0): Loading from cache
  - Installing symfony-cmf/routing-bundle (2.1.1): Loading from cache
  - Installing symfony-cmf/core-bundle (2.1.0): Loading from cache
  - Installing symfony/templating (v4.2.3): Loading from cache
  - Installing sonata-project/doctrine-extensions (1.1.5): Loading from cache
  - Installing twig/extensions (v1.5.4): Loading from cache
  - Installing sonata-project/datagrid-bundle (2.4.0): Loading from cache
  - Installing cocur/slugify (v3.2): Loading from cache
  - Installing sonata-project/core-bundle (3.15.1): Loading from cache
  - Installing sonata-project/cache (2.0.1): Loading from cache
  - Installing sonata-project/block-bundle (3.14.0): Loading from cache
  - Installing phpcr/phpcr (2.1.4): Loading from cache
  - Installing phpcr/phpcr-utils (1.4.0): Loading from cache
  - Installing jackalope/jackalope (1.3.7): Loading from cache
  - Installing jackalope/jackalope-doctrine-dbal (1.3.3): Loading from cache
  - Installing doctrine/data-fixtures (v1.3.1): Loading from cache
  - Installing doctrine/phpcr-odm (1.4.4): Loading from cache
  - Installing doctrine/phpcr-bundle (2.0.3): Loading from cache
  - Installing symfony-cmf/block-bundle (2.1.1): Loading from cache
  - Installing knplabs/knp-menu (2.3.0): Loading from cache
  - Installing knplabs/knp-menu-bundle (v2.2.1): Loading from cache
  - Installing symfony-cmf/menu-bundle (2.2.0): Loading from cache
  - Installing symfony-cmf/content-bundle (2.1.0): Loading from cache
  - Installing symfony-cmf/symfony-cmf (2.1.0): Loading from cache
  - Installing doctrine/phpcr-dbal-symfony-pack (1.0.1): Loading from cache
  - Installing symfony-cmf/cmf-phpcr-dbal-pack (1.0.0): Loading from cache
Writing lock file
Generating autoload files
ocramius/package-versions:  Generating version class...
ocramius/package-versions: ...done generating version class
Symfony operations: 13 recipes (942bf863c3f84d808d83cf96be6965a3)
  - Configuring symfony-cmf/routing-bundle (>=2.1.1): From auto-generated recipe
  - Configuring symfony-cmf/core-bundle (>=2.1.0): From auto-generated recipe
  - Configuring twig/extensions (>=1.0): From github.com/symfony/recipes:master
  - Configuring sonata-project/datagrid-bundle (>=2.4.0): From auto-generated recipe
  -  WARNING  sonata-project/core-bundle (>=3.9): From github.com/symfony/recipes-contrib:master
    The recipe for this package comes from the "contrib" repository, which is open to community contributions.
    Review the recipe at https://github.com/symfony/recipes-contrib/tree/master/sonata-project/core-bundle/3.9

    Do you want to execute this recipe?
    [y] Yes
    [n] No
    [a] Yes for all packages, only for the current installation session
    [p] Yes permanently, never ask again for this project
    (defaults to n): y
  - Configuring sonata-project/core-bundle (>=3.9): From github.com/symfony/recipes-contrib:master
  - Configuring sonata-project/block-bundle (>=3.14.0): From auto-generated recipe
  - Configuring doctrine/phpcr-bundle (>=2.0.3): From auto-generated recipe
  - Configuring symfony-cmf/block-bundle (>=2.1.1): From auto-generated recipe
  - Configuring knplabs/knp-menu-bundle (>=v2.2.1): From auto-generated recipe
  - Configuring symfony-cmf/menu-bundle (>=2.2.0): From auto-generated recipe
  - Configuring symfony-cmf/content-bundle (>=2.1.0): From auto-generated recipe
  -  WARNING  doctrine/phpcr-dbal-symfony-pack (>=1.0): From github.com/symfony/recipes-contrib:master
    The recipe for this package comes from the "contrib" repository, which is open to community contributions.
    Review the recipe at https://github.com/symfony/recipes-contrib/tree/master/doctrine/phpcr-dbal-symfony-pack/1.0

    Do you want to execute this recipe?
    [y] Yes
    [n] No
    [a] Yes for all packages, only for the current installation session
    [p] Yes permanently, never ask again for this project
    (defaults to n): y
  - Configuring doctrine/phpcr-dbal-symfony-pack (>=1.0): From github.com/symfony/recipes-contrib:master
  -  WARNING  symfony-cmf/cmf-phpcr-dbal-pack (>=1.0): From github.com/symfony/recipes-contrib:master
    The recipe for this package comes from the "contrib" repository, which is open to community contributions.
    Review the recipe at https://github.com/symfony/recipes-contrib/tree/master/symfony-cmf/cmf-phpcr-dbal-pack/1.0

    Do you want to execute this recipe?
    [y] Yes
    [n] No
    [a] Yes for all packages, only for the current installation session
    [p] Yes permanently, never ask again for this project
    (defaults to n): y
  - Configuring symfony-cmf/cmf-phpcr-dbal-pack (>=1.0): From github.com/symfony/recipes-contrib:master
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 255
!!  PHP Fatal error:  Uncaught Symfony\Component\Debug\Exception\FatalThrowableError: strlen() expects parameter 1 to be string, object given in /home/dmitry/Coding/php/phpcr-hello/vendor/sonata-project/block-bundle/src/DependencyInjection/Compiler/TweakCompilerPass.php:117
!!  Stack trace:
!!  #0 /home/dmitry/Coding/php/phpcr-hello/vendor/sonata-project/block-bundle/src/DependencyInjection/Compiler/TweakCompilerPass.php(46): Sonata\BlockBundle\DependencyInjection\Compiler\TweakCompilerPass->replaceBlockName(Object(Symfony\Component\DependencyInjection\ContainerBuilder), Object(Symfony\Component\DependencyInjection\Definition), 'cmf.block.actio...')
!!  #1 /home/dmitry/Coding/php/phpcr-hello/vendor/symfony/dependency-injection/Compiler/Compiler.php(95): Sonata\BlockBundle\DependencyInjection\Compiler\TweakCompilerPass->process(Object(Symfony\Component\DependencyInjection\ContainerBuilder))
!!  #2 /home/dmitry/Coding/php/phpcr-hello/vendor/symfony/dependency-injection/ContainerBuilder.php(748): Symfony\Component\DependencyInjection\Compiler\Compil in /home/dmitry/Coding/php/phpcr-hello/vendor/sonata-project/block-bundle/src/DependencyInjection/Compiler/TweakCompilerPass.php on line 117```

### PR DESCRIPTION
**Pull request by greg0ire**
## Subject

I am targeting this branch, because this is BC.

Closes #555

## Changelog

```markdown
### Fixed
- crash when using `null` as a block name in service definitions 
```

# How can I test this?

```shell
composer config repositories.greg0ire vcs https://github.com/greg0ire/SonataBlockBundle
composer require sonata-project/block-bundle "dev-fix-strlen as 3.14.0"
```

### CODE REVIEW

**Comment by greg0ire** [comment]
@doctorx32 @andsol @xleliberty @dpmR @muffe @peter-gribanov please test

File: src/DependencyInjection/Compiler/TweakCompilerPass.php [hunk 1]
```diff
@@ -114,7 +114,7 @@ private function replaceBlockName(ContainerBuilder $container, Definition $defin
         $arguments = $definition->getArguments();
 
         // Replace empty block id with service id
-        if (empty($arguments) || 0 === \strlen($arguments[0])) {
+        if (empty($arguments) || null === ($arguments[0]) || \is_string($arguments[0]) && 0 === \strlen($arguments[0])) {
```
Comment (core23) [inline src/DependencyInjection/Compiler/TweakCompilerPass.php]: Maybe you should introduce a special method for this check

Comment (greg0ire) [reply src/DependencyInjection/Compiler/TweakCompilerPass.php]: If you want, but I'm not really convinced… there is more distance now, and the comment already clarified what this was for.

**Comment by OskarStark** [comment]
Please check Travis

**Comment by greg0ire** [comment]
An unrelated, time-based test failed. I restarted the job.

**Comment by OskarStark** [comment]
@core23 please approve and merge

**Comment by muffe** [comment]
> @muffe please test
Fixes the error for me 👍

**Comment by greg0ire** [comment]
done

**Comment by kunicmarko20** [comment]
Thanks @greg0ire!

**Pull request merged**